### PR TITLE
FIR checker: introduce FunctionChecker alias

### PR DESCRIPTION
**Pull request by jsjeon**
and use it to add support diagnostic: FUNCTION_DECLARATION_WITH_NO_NAME

### CODE REVIEW

File: compiler/fir/analysis-tests/testData/resolve/diagnostics/conflictingOverloads.kt [hunk 1]
```diff
@@ -63,6 +63,6 @@ fun mest()
 
 class mest
 
-fun() {}
+<!FUNCTION_DECLARATION_WITH_NO_NAME{LT}!><!FUNCTION_DECLARATION_WITH_NO_NAME{PSI}!>fun()<!> {}<!>
```
Comment (jsjeon) [inline compiler/fir/analysis-tests/testData/resolve/diagnostics/conflictingOverloads.kt:66]: I'm afraid to say that positioning strategies for LT aren't working as expected here. Actually, I noticed this discrepancy while working on other diagnostics, which use new positioning strategies that I'm adding at that different work. I just wanted to make sure that using old, existing strategies match, but it's not the case here either.

It seems to me that many of tweaks made at https://github.com/JetBrains/kotlin/commit/e1802fde296dbdf0de43b76eedd0819836fd012c already indicated that LT positioning strategies may not be used at all. E.g.,
```kt
fun test(b: B) {
    b.<!HIDDEN{LT}!><!HIDDEN{PSI}!>foo<!>("")<!>
}
```
where PSI's report is rendered on the name `foo` only, while LT's report spans the whole call.

I can set a break on the checker and see the diagnostic is recorded for later rendering, but can't debug further whether or not LT positioning strategies (in particular, their `mark` overrides) are even invoked. We can move on while investigating why LT rendering differs, or I could wait. Or, you can let me know where to look at, e.g., test setup, etc.

Comment (mglukhikh) [reply compiler/fir/analysis-tests/testData/resolve/diagnostics/conflictingOverloads.kt:66]: Hope my fix from https://github.com/JetBrains/kotlin/pull/4041 will help...

**Review by mglukhikh — APPROVED** [review]

**Comment by mglukhikh** [comment]
Merged manually 👍

File: compiler/fir/analysis-tests/testData/resolve/diagnostics/conflictingOverloads.kt (hunk 1 above)
Comment (jsjeon) [reply compiler/fir/analysis-tests/testData/resolve/diagnostics/conflictingOverloads.kt:66]: Indeed! https://github.com/JetBrains/kotlin/commit/40bec30393dea6462ba2548da88296a80d99f15a#diff-34a9e16571c2d20587cba81a986efea888107e0a34d9c07a67401a72ba6157c3

**Pull request closed without merging**